app/models/foreman_openscap/policy.rb

```diff
@@ -168,8 +164,38 @@ def to_enc
       }.merge(period_enc)
     end
 
+    def should_validate?(step_i)
+      if new_record? && wizard_initiated?
+        step_index > step_i
+      elsif new_record? && !wizard_initiated?
+        true
+      else
+        !id.blank?
+      end
+    end
+
+    def wizard_initiated?
+      @wizard_initiated
+    end
+
+    def update_attributes(hash)
```

Comment:
also validation would work on self instead of hash, `erase_period_attrs` could the look like this

```ruby
def erase_period_attrs(attrs)
  attrs.each { |attr| self.public_send("#{attr}=", nil) }
end
```